### DISCUSSION TITLE
CUSTOM_LOOP_LIST not working in METPlus scripts

### DISCUSSION THREAD

**Discussion by lesnow93** [dtcenter/METplus#1877] [Configuration] [answered]
Hello again forum,

This is a follow up regarding issue #1838 (Running grid_stat on multiple models).  I tried using the CUSTOM_LOOP_LIST both as the high-level driver, and individually with PCP_COMBINE_CUSTOM_LOOP_LIST and GRID_STAT_CUSTOM_LOOP_LIST.  What I want to do is run pcp_combine, then grid_stat across multiple obs/fcst model combos (14 total "models").  I have attached my .conf and .log files here.  Thanks in advance for the help!
[metplus_201803291800.log](https://github.com/dtcenter/METplus/files/9823071/metplus_201803291800.log)
[metplus_multiModel.conf.txt](https://github.com/dtcenter/METplus/files/9823078/metplus_multiModel.conf.txt)

**Comment by CPKalb** [comment]
Hi:

In your configuration file, it looks like there are filename template tags in _DIR variables.  Specifically, FCST_PCP_COMBINE_INPUT_DIR and FCST_GRID_STAT_INPUT_DIR contain reference to the MODEL variable, which is set to {custom?fmt=%s}.  These template tags are not supported in _DIR variables, only in filename variables.  To fix this, you would want to move all reference tags from the _DIR variables to the _TEMPLATE variables.  Specifically:

Change FCST_PCP_COMBINE_INPUT_DIR = {INPUT_BASE}/servir_hkh/{MODEL}  to  FCST_PCP_COMBINE_INPUT_DIR = {INPUT_BASE}/servir_hkh, 

Also change FCST_PCP_COMBINE_INPUT_TEMPLATE= {init?fmt=%Y%m%d%H}/grib2/{MODEL}_{init?fmt=%y%m%d%H}00_wrfout_arw_d02.grb2f{lead?fmt=%HH}0000  to  FCST_PCP_COMBINE_INPUT_TEMPLATE= {MODEL}/{init?fmt=%Y%m%d%H}/grib2/{MODEL}_{init?fmt=%y%m%d%H}00_wrfout_arw_d02.grb2f{lead?fmt=%HH}0000

Does that fix the error you are currently seeing?

Christina

**Reply by georgemccabe** [reply]
To clarify, you can use the {MODEL} variable in _DIR variables if the value is a string. In your case MODEL is referencing the custom value via {custom?fmt=%s}. These filename template tags are not supported in the _DIR variables. In general, any part of the path that is constant can be put in the _DIR variable and anything that is dynamic or varies from run-to-run should go in the corresponding _TEMPLATE variable. In many cases you can also put the entire path in the _TEMPLATE variable and exclude the _DIR variable if that is easier for you.

**Reply by lesnow93** [reply]
Thanks for the help.  I adjusted my metplus config file and removed any references to dynamic settings in the _DIR variables.  I re-ran and metplus successfully runs one iteration of all model instances at the first time for pcp_combine, then errors out thereafter.  I also tried running ONLY pcp_combine and still the same error after the first time being processed.  I attached the new conf and log files.  Thanks!
[metplus_multiModel_APCP_v2.conf.txt](https://github.com/dtcenter/METplus/files/9841833/metplus_multiModel_APCP_v2.conf.txt)
[metplus_201803291800_v2.log](https://github.com/dtcenter/METplus/files/9841837/metplus_201803291800_v2.log)

**Comment by georgemccabe** [comment, marked as answer]
Hi @lesnow93, it looks like there is a bug in the logic that you have uncovered. I created an issue to fix it: https://github.com/dtcenter/METplus/issues/1884
It is likely a 1 line fix. Do you need this fix in a bugfix release for v4.1? The fix will be made for the upcoming v5.0.0 release.

**Reply by lesnow93** [reply]
Yes, if it's a quick fix, I'd like to drop in the modified file into my 4.1 build of METplus.  Thanks!

**Reply by georgemccabe** [reply]
Hi @lesnow93, you can apply the change from [this commit](https://github.com/dtcenter/METplus/commit/f16608befe5d4f302294b9e440d90be23fae940b) to your local copy of METplus v4.1.

**Reply by lesnow93** [reply]
Hello George.  Thanks very much for this bug fix!  I was successful in running metplus with pcp_combine+grid_stat using the custom string.

**Reply by georgemccabe** [reply]
Great to hear! I'm going to lock this discussion to prevent future posts, but feel free to create a new discussion if you have any other issues.
